src/csharp/Grpc.Core/Internal/NativeMetadataCredentialsPlugin.cs

```diff
@@ -78,7 +78,10 @@ private void NativeMetadataInterceptorHandler(IntPtr statePtr, IntPtr serviceUrl
             {
                 var context = new AuthInterceptorContext(Marshal.PtrToStringAnsi(serviceUrlPtr),
                                                          Marshal.PtrToStringAnsi(methodNamePtr));
-                StartGetMetadata(context, callbackPtr, userDataPtr);
+                // Don't await, we are in a native callback and need to return.
+                #pragma warning disable 4014
+                GetMetadataAsync(context, callbackPtr, userDataPtr);
```

Comment:
This seems a bit odd that we can't call this in an async way? Since IIC, this might do something like an oauth trip when called from core?